postgraphiql/src/components/PostGraphiQL.js

```diff
@@ -41,11 +41,21 @@ const defaultQuery = `\
 #
 `;
 
-const isSubscription = ({ query }) =>
-  parse(query).definitions.some(
+const isSubscription = ({ query, operationName }) => {
+  const { definitions } = parse(query);
+
+  if (operationName) {
+    const definition = definitions.find(def => def.name && def.name.value === operationName);
+    if (definition) {
+      return definition.operation === 'subscription';
+    }
+  }
+
+  return definitions.some(
     definition =>
       definition.kind === 'OperationDefinition' && definition.operation === 'subscription',
   );
+};
```

Comment:
My mistake, I forgot we had access to `graphql` here; we can just use the helper:

```js
import { getOperationAST, parse } from 'graphql';

const isSubscription = ({ query, operationName }) => {
  const operation = getOperationAST(parse(query), operationName);
  return operation.operation === 'subscription';
};
```